### PR TITLE
Hailemariam w2 databases

### PR DESCRIPTION
**Pull request by Hailemariam7**

### CODE REVIEW

File: Week2/assignment/authors.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+});
+
+connection.connect(function (err) {
+  if (err) throw err;
+  console.log("Connected to MySQL database");
+
+  connection.query("DROP TABLE IF EXISTS authors;");
+
+  const createTable =
+    "CREATE TABLE authors (author_id INT PRIMARY KEY AUTO_INCREMENT, author_name VARCHAR(50) NOT NULL, university VARCHAR(200), date_of_birth DATE NOT NULL, h_index INT, gender VARCHAR(1), mentor VARCHAR(50));";
```
Comment (codethenpizza) [inline Week2/assignment/authors.js:17]: Why did you add   `mentor VARCHAR` field?

File: Week2/assignment/queryEx3.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+connection.connect(function (err) {
+  if (err) {
+    console.error("Error connecting to database:", err);
+    return;
+  }
+
+  console.log("Connected to MySQL database");
+
+  // Query to print names of all authors and their corresponding mentors
+  const query1 = `
```
Comment (codethenpizza) [inline Week2/assignment/queryEx3.js:19]: Great effort on starting the query to print the names of all authors and their corresponding mentors. Your current query selects the author's name and the `mentor` field directly from the `authors` table, but to correctly achieve the goal of displaying each author's name along with their mentor's name, we need to join the `authors` table with itself. The mentor's name is stored as a field in the `authors` table, which requires a self-join to fetch it correctly. 
This will give you a list of authors along with their mentors' names. Give it a try and see if this helps you get the correct result. You're almost there!

Keep up the good work!

**Review by codethenpizza — COMMENTED** [review]
Good job! 
There's only a few things to improve

File: Week2/assignment/authors.js (hunk 1 above)
Comment (Hailemariam7) [reply Week2/assignment/authors.js:17]: Because that's how i understood the question, which says as follows. [https://github.com/HackYourAssignment/databases-cohort47/blob/main/Week2/MAKEME.md#:~:text=Write%20a%20query%20that%20adds%20a%20column%20called%20mentor%20to%20authors%20table%20that%20references%20the%20column%20author_id.%20For%20integrity%20add%20a%20foreign%20key%20on%20this%20column.](week2-assignment). What was the right thing to do?

File: Week2/assignment/queryEx3.js (hunk 2 above)
Comment (Hailemariam7) [reply Week2/assignment/queryEx3.js:19]: I have corrected it. Thank you!

**Comment by Hailemariam7** [comment]
Thank you for the review! I have corrected the code according to the correction.

File: Week2/assignment/authors.js (hunk 1 above)
Comment (codethenpizza) [reply Week2/assignment/authors.js:17]: I see, the thing is - you don't need an actual column, but it should be a reference to another author.

you basically did it in the `addColumn` query. So anytime you will want to query mentor you don't query mentor as a column, but do a join within the table and get the author ID by the reference. Mentor ID will allow you to get the mentor's name which is in our case same as the author's name.

does it make sense for you?

Comment (codethenpizza) [reply Week2/assignment/authors.js:17]: oh, you updated it correctly in exercise 3.  As you may see, you used relation and not the actual mentor field

**Comment by codethenpizza** [comment]
Look good to me!

**Review by codethenpizza — APPROVED** [review]

**Comment by Hailemariam7** [comment]
Thanks a lot!

File: Week2/assignment/authors.js (hunk 1 above)
Comment (Hailemariam7) [reply Week2/assignment/authors.js:17]: yes, it makes total sense yeah.
